x-pack/plugins/ml/public/settings/calendar/edit/new_event_modal.js

```diff
@@ -0,0 +1,296 @@
+    const endMoment = moment(date);
+
+    start = startMoment.startOf('day');
+
+    if (start > end) {
+      end = endMoment.startOf('day').add(1, 'days');
+    }
+    this.setState({
+      startDate: start,
+      endDate: end,
+      startDateString: start.format(TIME_FORMAT),
+      endDateString: end.format(TIME_FORMAT)
+    });
+  }
+
+  handleChangeEnd = (date) => {
+    let start = this.state.startDate;
+    let end = null;
+
+    const startMoment = moment(date);
+    const endMoment = moment(date);
+
+    end = endMoment.startOf('day');
+
+    if (start > end) {
+      start = startMoment.startOf('day').subtract(1, 'days');
+    }
+    this.setState({
+      startDate: start,
+      endDate: end,
+      startDateString: start.format(TIME_FORMAT),
+      endDateString: end.format(TIME_FORMAT)
+    });
+  }
+
+  handleTimeStartChange = (event) => {
+    const dateString = event.target.value;
+    let isValidDate = false;
+
+    if (dateString.length === VALID_DATE_STRING_LENGTH) {
+      isValidDate = moment(dateString).isValid(TIME_FORMAT, true);
+    } else {
+      this.setState({
+        startDateString: dateString,
+      });
+    }
+
+    if (isValidDate) {
+      this.setState({
+        startDateString: dateString,
+        startDate: moment(dateString)
+      });
+    }
+  }
+
+  handleTimeEndChange = (event) => {
+    const dateString = event.target.value;
+    let isValidDate = false;
+
+    if (dateString.length === VALID_DATE_STRING_LENGTH) {
+      isValidDate = moment(dateString).isValid(TIME_FORMAT, true);
+    } else {
+      this.setState({
+        endDateString: dateString,
+      });
+    }
+
+    if (isValidDate) {
+      this.setState({
+        endDateString: dateString,
+        endDate: moment(dateString)
+      });
+    }
+  }
+
+  renderRangedDatePicker = () => {
+    const {
+      startDate,
+      endDate,
+      startDateString,
+      endDateString,
+    } = this.state;
+
+    const timeInputs = (
+      <Fragment>
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFormRow label="From:" helpText={TIME_FORMAT}>
+              <EuiFieldText
+                name="startTime"
+                onChange={this.handleTimeStartChange}
+                placeholder={TIME_FORMAT}
+                value={startDateString}
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiFormRow label="To:" helpText={TIME_FORMAT}>
+              <EuiFieldText
+                name="endTime"
+                onChange={this.handleTimeEndChange}
+                placeholder={TIME_FORMAT}
+                value={endDateString}
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </Fragment>
+    );
+
+    return (
+      <Fragment>
+        <EuiSpacer size="s" />
+        {timeInputs}
+        <EuiSpacer size="s" />
+        <EuiFormRow fullWidth>
+          <EuiDatePickerRange
```

Comment:
Is this component fully accessible as used here? I wasn't able to navigate to particular dates

Reply:
All the inputs and the arrows to move between months are - this should give keyboard access to all functionality. Moving around through each day in the calendars themselves looks like it isn't - not sure we can override that on our side, though. 🤔